server-spi/src/main/java/org/keycloak/models/RealmProvider.java

```diff
@@ -176,7 +206,7 @@ default ClientModel addClient(RealmModel realm, String clientId) {
     /**
      * @deprecated Use the corresponding method from {@link ClientProvider}. */
     @Override
-    public List<ClientModel> searchClientsByClientId(RealmModel realm, String clientId, Integer firstResult, Integer maxResults);
+    public Stream<ClientModel> searchClientsByClientIdStream(RealmModel realm, String clientId, Integer firstResult, Integer maxResults);
```

Comment:
In case I revert this I have to implement the method in `JpaRealmProvider` and `RealmCacheSession` or the method has to be removed from `RealmProvider`.

Reply:
removed